spec/Section 3 -- Type System.md

```diff
@@ -1052,8 +1051,7 @@ Interfaces are never valid inputs.
 
 Interface types have the potential to be invalid if incorrectly defined.
 
-1. An Interface type must define one or more fields.
```

Comment:
Interesting discussion around this point specifically:

https://github.com/graphql/graphql-spec/issues/236

> This also drives the requirement that Interfaces must define at least one field - otherwise the Interface would not actually describe a contract to be fulfilled. This concept of contract-less interfaces is often called "marker interfaces" and in my opinion (hopefully not an uncommon opinion) are a bad practice borrowed from limited object-oriented languages. 
> -- @leebyron 

---

I'm personally in favour of field-less interfaces as a placeholder for adding shared fields later. E.g. you might say that 3 different types are `Searchable` and in future you may require that they have `name` and `url` in common so that you can display them in a search results page even if you don't have specific handling for some of the concrete types, but when you start designing your schema you might not know what those "Searchable" shared fields are/should be named yet.

I also think this can be useful for [interfaces that implement other interfaces](https://github.com/graphql/graphql-spec/pull/373):

```graphql
interface Searchable implements Node {}
```

e.g. to say that everything that's searchable is a Node, but not everything that's a Node is searchable.

Reply:
Based on the latest from the transitive interface implementation RFC

`interface Searchable implements Node {}`

would actually be an error, unless Node itself did not have a contract with any fields. `implements` instead of `extends` is intentional here, so `Searchable` would have to implement (declare) all fields `Node`'s contract requires.